module-system/module-implementations/sov-evm/src/query.rs

```diff
@@ -212,13 +212,21 @@ impl<C: sov_modules_api::Context> Evm<C> {
         unimplemented!("eth_sendTransaction not implemented")
     }
 
-    // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/502
     #[rpc_method(name = "blockNumber")]
     pub fn block_number(
         &self,
```

Comment:
Let's test it in a similar way: https://github.com/Sovereign-Labs/sovereign-sdk/pull/886#discussion_r1327209827

Reply:
Actually we might need to wait until #870 is merged with this particular test. LGTM